src/support.cpp

```diff
@@ -1479,30 +1479,30 @@ std::pair<Polygons, Polygons> AreaSupport::computeBasicAndFullOverhang(const Sli
     const Polygons outlines = mesh.layers[layer_idx].getOutlines();
     constexpr bool no_support = false;
     constexpr bool no_prime_tower = false;
-    const Polygons outlines_below = storage.getLayerOutlines(layer_idx - 1, no_support, no_prime_tower);
 
-    // To avoids generating support for textures on vertical surfaces, a moving average
-    // is taken over smooth_height. The smooth_height is currently an educated guess
-    // that we might want to expose to the frontend in the future.
     constexpr double smooth_height = 0.4; //mm
-    const auto layers_above = static_cast<LayerIndex>(std::round(smooth_height / mesh.settings.get<double>("layer_height")));
-    Polygons outlines_above;
-    for (int i = 0; layer_idx + i < mesh.layers.size() && i <= layers_above; ++i)
-    {
-        outlines_above.add(storage.getLayerOutlines(layer_idx + i, no_support, no_prime_tower));
-    }
-    outlines_above = outlines_above.unionPolygons();
+    const auto layers_below = static_cast<LayerIndex>(std::round(smooth_height / mesh.settings.get<double>("layer_height")));
 
     const coord_t layer_height = mesh.settings.get<coord_t>("layer_height");
     const AngleRadians support_angle = mesh.settings.get<AngleRadians>("support_angle");
     const double tan_angle = tan(support_angle) - 0.01; // The X/Y component of the support angle. 0.01 to make 90 degrees work too.
+    // overhang areas protruding less then `max_dist_from_lower_layer` don't need support
     const coord_t max_dist_from_lower_layer = tan_angle * layer_height; // Maximum horizontal distance that can be bridged.
 
+    // To avoids generating support for textures on vertical surfaces, a moving average
+    // is taken over smooth_height. The smooth_height is currently an educated guess
+    // that we might want to expose to the frontend in the future.
+    Polygons outlines_below = storage.getLayerOutlines(layer_idx - 1, no_support, no_prime_tower)
+                                                         .offset(max_dist_from_lower_layer);
```

Comment:
I'm not really a fan of these huge amounts of spaces and just arbitrarily placing the next bit vaguely under the last.

If you need multiple lines I'd say do something like this:

~~~
Type varname =
    blahblah
    .funky_name_name(aaaa_blahbloehbleh)
    .smurfsmurfsmurf(yolo - 1);
~~~

(For completeness sake, and this isn't really addressed to you in particular: This goes for those functions with lots of parameters as well that I now see all over the place. Place even the first parameter on a new line with a single added indentation instead of having the first argument on the same line as the function and then having a huge arbitrarily long indentation for each next line to line it out with the previous parameter...)

Reply:
https://github.com/Ultimaker/CuraEngine/pull/1848/commits/fbd017f54593c4f4340e6b0b41a6d9fdfe1c876a